corehq/apps/reports/filters/api.py

```diff
@@ -154,10 +154,36 @@ def extra_data_sources(self):
         # extra data sources to be included for filtering
         raise NotImplementedError('Not implemented yet')
 
+    def custom_locations_search(self):
+        """
+        When the query is specifically searching for just locations and not any other entity like user, group.
+        For ex: enter "Bihar/patn" would match child locations under locations named Bihar, having name like
+        patn.
+        """
+        return '/' in self.q
+
+    @staticmethod
+    def _get_location_specific_custom_filters(query):
+        query_sections = query.split("/")
```

Comment:
Looks like text after the _second_ slash will be discarded. Not sure if you want to support arbitrary parent/child/grandchild searches, or if you just want to handle the first slash? If the latter, consider changing this to `query.split("/", 1)`, which will only split on the first `/`.

Reply:
Sorry not sure i understand. Sharing what i intended.
Just expecting one "/" in the query. So "bihar/patna" would give parent_name as bihar and patna as the search query for the descendants.

Reply:
but what happens if someone types "bihar/patna/danapur"? I believe currently "danapur" would be discarded/ignored.

Reply:
ya thats fine.
There is no reason for someone to search like this.
to search `danapur` one can just type `bihar/danapur`

Reply:
some people may not realize that, and I don't think it's a good idea to be silently ignoring part of what they typed

Reply:
What about something like this for `filter_path_by_user_input`:
```python
def filter_path_by_user_input(self, domain, user_input):
    query = self
    for part in filter(None, user_input.split('/')):
        if part.startswith('"') and part.endswith('"'):
            direct_matches = query.filter(name__iexact=part[1:-1])
        else:
            direct_matches = query.filter_by_user_input(domain, part)
        query = self.get_queryset_descendants(direct_matches, include_self=True)
    return query
```
This is an expansion of my earlier suggestion - it does partial matches on as many nodes as are entered, and if you wrap any node in quotes, it'll do an exact match on that node.  That will allow you to do exact matches also on the terminal node.  It sounds like we still all disagree on how to specify exact matches, so I'm just throwing my hat into the ring.

Reply:
Thank you @esoergel. That worked well in terms of search.

For some edge case it *might* show duplicates. I am not sure but saw some duplicates when searching on local system. Could not replicate.
But,
At the back ES Search complaints about `"` and '\' in search so would lead to errors whenever the search is done for input while the user is typing the complete string.

Reply:
> At the back ES Search complaints about `"` and `'` in search

Are they being escaped properly? If yes, and those characters are truly not allowed in ES search strings (that would be surprising to me) we could strip unbalanced quotes.

Reply:
Actually, on second though, I think the way Google handles it is if there is an unbalanced quote character then everything from that character to the end of the string is matched exactly. In other words, an implicit balancing quote is added at the end of the string. We should probably do that.

Reply:
I like the idea of an implicit balancing quote - that sounds like a good idea.  Manish is describing ES errors, which are probably occurring in the other data sources that are searched in this filter.  If there's a slash, it's clear that the user is searching for a location, so otherwise we can't tell.  On the ES-backed data sources, quotes get interpreted as "advanced" search queries [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/queries.py#L56-L85).

I just took a look at the docs for [query string queries](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-query-string-query.html#query-dsl-query-string-query) (which we use to support things like quotes and wildcards), and I noticed an alternative, the [simple query string query](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-simple-query-string-query.html#query-dsl-simple-query-string-query):
> A query that uses the SimpleQueryParser to parse its context. Unlike the regular query_string query, the simple_query_string query will never throw an exception, and discards invalid parts of the query.

That sounds like it would solve the problem, and it seems like we should probably be using that for user-inputted searches anyways.

Manish, I'm sorry for how much this feature and discussion has ballooned, feel free to define the scope wherever seems appropriate.

Reply:
Sure Ethan.
I am going to keep it the way it is as of now with the original idea.
Feel free to make improvements later.